src/rosinstall_generator/distro.py

```diff
@@ -132,26 +133,38 @@ def _generate_rosinstall_for_package(distro, pkg_name, flat=False, tar=False):
 
 
 def _generate_rosinstall(local_name, url, release_tag, tar=False, vcs_type=None):
+    logger = logging.getLogger('rosinstall_generator.generate')
+
     if tar:
-        # the repository name might be different than repo.name coming from rosdistro
-        repo_name = os.path.basename(url[:-4])
-        suffix = '.git'
-        if not url.endswith(suffix):
-            raise RuntimeError("The repository '%s' must end with '%s': %s" % (repo_name, suffix, url))
-        url = url[:-len(suffix)] + '/archive/{0}.tar.gz'.format(release_tag)
-        data = [{
-            'tar': {
-                'local-name': local_name,
-                'uri': url,
-                'version': '{0}-{1}'.format(repo_name, release_tag.replace('/', '-'))
-            }
-        }]
-    else:
-        data = [{
-            vcs_type or 'git': {
-                'local-name': local_name,
-                'uri': url,
-                'version': release_tag
+        # Github tarball:    https://github.com/ros/ros_comm/archive/1.11.20.tar.gz
+        # Bitbucket tarball: https://bitbucket.org/osrf/gazebo/get/gazebo7_7.3.1.tar.gz
+        # Gitlab tarball:    https://gitlab.com/gitlab-org/gitlab-ce/repository/archive.tar.gz?ref=master
+        match = re.match('(?:https?:\/\/|ssh:\/\/|git:\/\/|git@)([\w.-]+)[:/]([\w/-]*)(?:\.git)?$', url)
```

Comment:
There's some logic which does similar things in pip:

https://github.com/pypa/pip/blob/3402e3892e188a7117a954b0ac6adb6183bbd622/pip/vcs/git.py#L25-L32
https://github.com/pypa/pip/blob/3402e3892e188a7117a954b0ac6adb6183bbd622/pip/vcs/git.py#L171-L186

Kind of blah, though. Really all I want is the server and path; I feel like a regex should be a fairly easy way of getting this. If nothing else, I can simplify the regex to make the scheme more generic, eg:

    match = re.match('(?:\w+:\/\/|git@)([\w.-]+)[:/]([\w/-]*)(?:\.git)?$', url)

The `re.match` function will only return matches at the beginning of the input anyway, so a ^ is superfluous, eg:

```
$ python
>>> import re
>>> bool(re.match('foo', 'oo'))
False
```